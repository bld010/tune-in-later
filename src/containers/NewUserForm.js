import React, { Component } from 'react';
import './NewUserForm.scss';

class NewUserForm extends Component {
  // constructor(props) {
  //   super (props),
  //   this.state = {
  //     email: '',
  //     password: ''
  //   }
  // }

  handleSubmitNewUser = (e) => {
    // e.preventDefault;
    //check if email exists?
    //set user to selecteduser
    //hide the login/new user sections
  }

  render() {
    return(
      <form class="NewUserForm">
        <h2>New here? Create an account below:</h2>
        <input 
          name="new-user-name"
          type="text"
          placeholder="Enter your name"
          ></input>
        <input 
          name="new-user-email" 
          type="text"
          placeholder="Enter your e-mail"></input>
        <input 
          name="new-user-password" 
          type="text"
          placeholder="Enter a password"
          ></input>
          <button type="submit">Create Account</button>
      </form>
    )
  }
}

export default NewUserForm;