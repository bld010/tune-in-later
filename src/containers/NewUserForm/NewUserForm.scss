.NewUserForm {
  display: flex;
  flex-direction: column;
  align-items: center;

  & button {
  height: 30px;
  width: 100px;
  border: 1px solid snow;
  border-radius: 8px;

    &:hover,
    &:focus {
      font-weight: bold;
      font-style: italic;
      color: #353EBA;
      border: 1.5px solid #353EBA;
    }

  }
  
  & button {
    width: 200px;
  }
  
}