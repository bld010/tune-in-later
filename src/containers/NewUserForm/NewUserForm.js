import React, { Component } from 'react';
import { connect } from 'react-redux';
import './NewUserForm.scss';
import { setCurrentUser } from '../../actions';
import PropTypes from 'prop-types';


class NewUserForm extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      error: ''
    }
  }

  addNewUserFetch = () => {

    const body = {
      name: this.state.name,
      email: this.state.email,
      password: this.state.password
    }

    const options = {
      method: "POST",
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(body)
    }

    fetch('http://localhost:3001/api/v1/users', options)
      .then(resp => {
        if (!resp.ok) {
          throw Error('Something went wrong');
        }
        return resp.json()
      })
      .then(data => console.log(data))
      .catch(err => 
        this.setState({
        name: '',
        email: '',
        password: '',
        error: err.message})
        )
  }


  handleNewUserInputs = (e) => {
    this.setState({ [e.target.name]: e.target.value})
  }

  clearNewUserInputs = () => {
    this.setState({
      name: '',
      email: '',
      password: '',
      error: ''
    })
  }

  handleSubmitNewUser = (e) => {
    e.preventDefault();
    this.addNewUserFetch();
    this.props.setCurrentUser({
      name: this.state.name,
      email: this.state.email,
      password: this.state.password
    });
    this.clearNewUserInputs();
  }

  render() {
    return(
      <form className="NewUserForm">
        <h2>New here? Create an account below:</h2>
        <input 
          name="name"
          type="text"
          value={this.state.name}
          onChange={this.handleNewUserInputs}
          placeholder="Enter Your Name"
          ></input>
        <input 
          name="email" 
          type="text"
          value={this.state.email}
          onChange={this.handleNewUserInputs}
          placeholder="Enter Your Email"></input>
        <input 
          name="password" 
          type="password"
          value={this.state.password}
          placeholder="Enter A Password"
          onChange={this.handleNewUserInputs}
          ></input>
          {this.state.error && <p>Sorry, that e-mail is already associated with an account.</p>}
          <button type="submit" onClick={this.handleSubmitNewUser}>Create Account</button>
      </form>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user))
})

export default connect(null, mapDispatchToProps)(NewUserForm);

NewUserForm.propTypes = {
  setCurrentUser: PropTypes.func.isRequired
}