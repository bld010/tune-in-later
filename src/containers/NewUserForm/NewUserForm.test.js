import React from 'react';
import { shallow } from 'enzyme';
import { NewUserForm } from './NewUserForm';

describe('NewUserForm', () => {

  let wrapper; 
  let mockSetCurrentUser;

  beforeEach(() => {

    mockSetCurrentUser = jest.fn();

    wrapper = shallow(<NewUserForm 
      setCurrentUser={mockSetCurrentUser}
    />)
  })

  it('should match the snapshot with correct data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })

 
})