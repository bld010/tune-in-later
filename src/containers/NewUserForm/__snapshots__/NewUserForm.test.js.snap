// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewUserForm should match the snapshot with correct data passed in 1`] = `
<form
  className="NewUserForm"
>
  <h2>
    New here? Create an account below:
  </h2>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Enter your name"
    type="text"
    value=""
  />
  <input
    name="email"
    onChange={[Function]}
    placeholder="Enter your e-mail"
    type="text"
    value=""
  />
  <input
    name="password"
    onChange={[Function]}
    placeholder="Enter a password"
    type="password"
    value=""
  />
  <button
    onClick={[Function]}
    type="submit"
  >
    Create Account
  </button>
</form>
`;
